swan-lake/learn-the-platform/ballerina-persist/supported-data-stores.md

```diff
@@ -0,0 +1,191 @@
+---
+layout: ballerina-persist-documentation-left-nav-pages-swanlake
+title: Supported Data Stores
+description: The sections gives details on the Supported Data Stores.
+keywords: ballerina, programming language, ballerina packages, persist, data store, in-memory, mysql, google sheets
+permalink: /learn/ballerina-persist/supported-datastores/
+active: persist_supported_datastores
+intro: In the realm of `bal persist` feature, selecting the right data store is crucial for ensuring the reliable storage and retrieval of data. Various data stores exist, each with its strengths and weaknesses. The `bal persist` currently supports three data stores. Depending on the data store you selected for your application, configuration and supported types are varied.
+redirect_from:
+- /learn/ballerina-persist/supported-datastores/
+---
+
+Following are the data stores supported,
+
+* In-Memory
+* MySQL
+* Google Sheets [Experimental]
+
+The datastore is passed as a parameter to the `persist init` command. If you do not specify a data store, the In-Memory data store is used by default. 
+
+```shell
+$ bal persist init --datastore mysql
+```
+
+It is recorded in the `Ballerina.toml` file in your project as follows:
+
+```toml
+[persist]
+datastore = mysql
+...
+```
+The following sections describe the configuration and supported types for each data store.
+
+## In-Memory
+
+The In-Memory data store is a simple data store that stores data in memory. This data store is useful for testing purposes. The In-Memory data store is the default data store for `bal persist`. Therefore, you do not need to explicitly specify the data store when you are using the In-Memory data store.
+
+### Supported Ballerina Types
+
+The In-Memory data store supports the following Ballerina types:
+
+* [any](https://ballerina.io/learn/language-basics/#any-type)
+* error
+
+### Configuration
+
+The In-Memory data store does not require any configuration. 
+
+## MySQL
+
+The MySQL data store is a relational database management system that stores data in tables. The MySQL data store is useful for storing data in a relational format. The MySQL data store is not the default data store for `bal persist`. Therefore, you need to explicitly specify the data store when initializing `bal persist` in your application. like below,
+
+```shell
+$ bal persist init --datastore mysql
+```
+
+### Supported Ballerina Types
+
+The following table lists the Ballerina types supported by the MySQL data store and the corresponding SQL types used to store the data in the database.
+
+|  Ballerina Type  |    SQL Type     |
+|:----------------:|:---------------:|
+|       int        |       INT       |
```

Comment:
```suggestion
|       int        |       INT       |
```
Shall we add backticks to all these values in both columns?

Reply:
No need.